src/nsrt_learning/side_predicate_learning.py

```diff
@@ -456,18 +479,67 @@ def _sideline(self) -> List[PartialNSRTAndDatastore]:
                     for a in pnad.op.preconditions
                 }
 
+        # Now that the add effects and preconditions are correct,
+        # make a list of all final PNADs. Note
+        # that these final PNADs only come from the
+        # param_opt_to_nec_pnads dict, since we can be assured
+        # that our backchaining process ensured that the
+        # PNADs in this dict cover all of the data!
+        all_pnads = []
+        for pnad_list in param_opt_to_nec_pnads.values():
```

Comment:
i'm nervous about nondeterministic ordering here... is this dictionary always built in the same way? if we're not sure, we can sort this to ensure determinism

Reply:
Let's sort just to be safe